models/DataObject/Classificationstore/CollectionConfig.php

```diff
@@ -262,4 +323,64 @@ public function setStoreId($storeId)
     {
         $this->storeId = $storeId;
     }
+
+    /**
+     * Set cache item for a given cache key
+     *
+     * @param CollectionConfig $config
+     * @param string $cacheKey
+     */
+    private static function setCache(CollectionConfig $config, string $cacheKey): void
+    {
+        if (self::$cacheEnabled) {
+            Cache\Runtime::set($cacheKey, $config);
+        }
+
+        Cache::save($config, $cacheKey, [], null, 0, true);
+    }
+
+    /**
+     * Remove a cache item for a given cache key
+     *
+     * @param string $cacheKey
+     */
+    private static function removeCache(string $cacheKey): void
+    {
+        Cache::remove($cacheKey);
+        Cache\Runtime::set($cacheKey, null);
+    }
+
+    /**
+     * Get a cache item for a given cache key
+     *
+     * @param string $cacheKey
+     *
+     * @return CollectionConfig|bool
+     */
+    private static function getCache(string $cacheKey): CollectionConfig|bool
+    {
+        if (self::$cacheEnabled && Cache\Runtime::isRegistered($cacheKey) && $config = Cache\Runtime::get($cacheKey)) {
+            return $config;
+        }
+
+        return Cache::load($cacheKey);
+    }
```

Comment:
To avoid code duplication and make it easier to maintain, we should put that code into a trait. Since this is very generic, i think we can put this trait here: `lib/Cache/RuntimeCacheTrait.php`. 
As a follow up we could then also migrate existing usages of the runtime cache to this trait.

Reply:
Done...